### DISCUSSION TITLE
FCM data message seems not to work

### DISCUSSION THREAD

**Discussion by dayclocks** [ionic-team/capacitor#3663] [Q&A]
Hi, 

FCM Notification messages are working perfectly fine. But what about FCM data messages? They seem not to work (iOS). Can anyone help me?

**Comment by kevinboosten** [comment]
Hi @dayclocks! 

This is by design. Capacitor is not handling remote notifications for you as stated here by @jcesarmobile: https://github.com/ionic-team/capacitor/issues/3736#issuecomment-717823916 

Also see this issue: #2401 and the potential way of implementing data notifications https://github.com/ionic-team/capacitor/issues/2401#issuecomment-633285481.
